network/netcore/src/transport/tcp.rs

```diff
@@ -88,18 +87,22 @@ impl Transport for TcpTransport {
         Ok((
             TcpListenerStream {
                 inner: listener,
-                config,
+                config: self.clone(),
             },
             actual_addr,
         ))
     }
 
     fn dial(&self, addr: NetworkAddress) -> Result<Self::Outbound, Self::Error> {
-        let socket_addr = multiaddr_to_string(&addr)?;
-        let config = self.clone();
+        let (hostport, _addr_suffix) =
+            parse_hostport(addr.as_slice()).ok_or_else(|| invalid_addr_error(&addr))?;
```

Comment:
btw, at this point addr is still an untrusted input right?

Reply:
yep

Reply:
not too sure what to do with this, but a few suggestions:

* add a comment for reviewers to understand that this is untrusted, and to pay attention to the parsing
* change the name of the variable to reflect that it is untrusted
* create an Untrusted type wrapping the correct type